app/controllers/sessions_controller.rb

```diff
@@ -1,20 +1,56 @@
 class SessionsController < Clearance::SessionsController
+  before_action :user, only: :create
+
   def create
-    @user = find_user(params.require(:session))
+    if verifying_otp? && !@user.otp_verified?(session_params[:otp])
+      session[:mfa_user] = nil
+      login_failure(t('two_factor_auths.incorrect_otp'))
+    elsif !verifying_otp? && @user&.mfa_enabled?
+      session[:mfa_user] = @user.id
+      render 'sessions/otp_prompt'
+    else
+      do_login
+    end
+  end
```

Comment:
This if else logic could be simpler if `'sessions/otp_prompt'` made request to a new method like `mfa_create`.

Reply:
yeah. and in `mfa_create` check `session[:mfa_user]` seems can achieve the same functionality